src/gatsby-node.js

```diff
@@ -0,0 +1,569 @@
+        try {
+          copyChunk = copyChunk
+            .replace(regex, "")
+            .replace(brailleRegex, "")
+            .trimStart();
+
+          if (Object.keys(ALREADY_LOGGED).includes(copyChunk)) {
+            if (ALREADY_LOGGED[copyChunk]) {
+              return originalStdoutWrite(chunk, encoding, callback);
+            } else {
+              ALREADY_LOGGED[copyChunk] = true;
+            }
+          }
+
+          let deletedPages = deletedPagesRegex.exec(copyChunk);
+          let changedPages = changedPagesRegex.exec(copyChunk);
+
+          if (deletedPages) {
+            DELETED_PAGES = deletedPages[1];
+          }
+
+          if (changedPages) {
+            CHANGED_PAGES = changedPages[1];
+          }
+
+          if (copyChunk.includes(clearingCache)) {
+            CLEARING_CACHE = true;
+          }
+
+          if (copyChunk !== "") {
+            winstonLogger.log({
+              level: "info",
+              message: copyChunk,
+            });
+          }
+        } catch (err) {
+          winstonLogger.log({
+            level: "error",
+            message: err.message,
+          });
+        }
+      }
+
+      return originalStdoutWrite(chunk, encoding, callback);
+    };
+
+    console.error = function (msg) {
+      winstonLogger.log({
+        level: "error",
+        message: msg,
+      });
+    };
+
+    console.warn = function (msg) {
+      //
+      winstonLogger.log({
+        level: "warn",
+        message: msg,
+      });
+    };
+  }
+}
+
+const bootstrapTime = performance.now();
+let lastApi; // Current benchmark state, if any. If none then create one on next lifecycle.
+
+let benchMeta;
+let nextBuildType = process.env.BENCHMARK_BUILD_TYPE ?? `initial`;
+
+function reportInfo(...args) {
+  (lastApi ? lastApi.reporter : console).info(...args);
+}
+
+function reportError(...args) {
+  (lastApi ? lastApi.reporter : console).error(...args);
+}
+
+class BenchMeta {
+  constructor() {
+    this.flushing = undefined; // Promise of flushing if that has started
+
+    this.flushed = false; // Completed flushing?
+
+    this.localTime = new Date().toISOString();
+    this.timestamps = {
+      // TODO: we should also have access to node's timing data and see how long it took before bootstrapping this script
+      bootstrapTime,
+      // Start of this file
+      instantiationTime: performance.now(),
+      // Instantiation time of this class
+      benchmarkStart: 0,
+      // Start of benchmark itself
+      preInit: 0,
+      // Gatsby onPreInit life cycle
+      preBootstrap: 0,
+      // Gatsby onPreBootstrap life cycle
+      preBuild: 0,
+      // Gatsby onPreBuild life cycle
+      postBuild: 0,
+      // Gatsby onPostBuild life cycle
+      benchmarkEnd: 0, // End of benchmark itself
+    };
+    this.started = false;
+  }
+
+  getMetadata() {
+    let siteId = ``;
+
+    try {
+      siteId =
+        JSON.parse(process.env?.GATSBY_TELEMETRY_TAGS ?? `{}`)?.siteId ?? ``;
```

Comment:
i mentioned this in the zoom. optional chaining is not available for the version of node we use, so we need to account for that in some way.

there may be other such things im not aware of 🙈  this is the most common one i run into since the sites use node 12.

Reply:
great catch i'm using Node 14 so this is definitely something we'll need to account for